code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -36,20 +36,25 @@
 	flags_pass = PASSTABLE
 	use_plasma(10)
 	throw_at(T, 6, 2, src) //Victim, distance, speed
-	spawn(6)
-		if(!xeno_caste.hardcore)
-			flags_pass = initial(flags_pass) //Reset the passtable.
-		else
-			flags_pass = 0 //Reset the passtable.
-
-	spawn(xeno_caste.pounce_delay)
-		usedPounce = 0
-		to_chat(src, "<span class='notice'>You get ready to pounce again.</span>")
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
```

Comment:
No path needed on src

Reply:
Also, why 2 timers if it was originally 1 spawn and why does one have a magic number delay

Reply:
it was originally two, from what i understand the spawn(6) reset_flags() is to make it possible to pass through window frames/tables etc when pouncing/thrown but not having the effect last for the whole of the pounce cooldown duration